bench/wtperf/wtperf.c

```diff
@@ -131,6 +146,173 @@ randomize_value(CONFIG_THREAD *thread, char *value_buf)
 }
 
 static int
+setup_truncate(
+    CONFIG *cfg, CONFIG_THREAD *thread,
+    TRUNCATE_CONFIG *trunc, WT_CURSOR *cursor) {
+
+	int ret;
+	size_t i;
+	uint64_t end_point_val, final_milestone_gap, start_point_val;
+	char *key, *truncate_key;
+	TRUNCATE_QUEUE_ENTRY *truncate_item;
+
+	ret = 0;
+	end_point_val = final_milestone_gap = start_point_val = 0;
+
+	/* Truncation percentage value. eg 10% is 0.1 */
+	trunc->truncation_percentage =
+	    (double)thread->workload->truncate_pct / 100;
+	/* How many entries between each milestone */
+	trunc->truncate_milestone_gap =
+	    thread->workload->truncate_count * trunc->truncation_percentage;
+	/* How many milestones we need */
+	trunc->needed_milestones = thread->workload->truncate_count /
+	    trunc->truncate_milestone_gap;
+
+	final_milestone_gap = trunc->truncate_milestone_gap;
+
+	/* Reset this value for use again */
+	trunc->truncate_milestone_gap = 0;
+
+	/*
+	 * Here we check if there is data in the collection. If there is
+	 * data available, then we need to setup some initial truncation
+	 * milestones.
+	 */
+	ret = cursor->next(cursor);
+
+	if (ret == 0) {
+		ret = cursor->get_key(cursor, &key);
+	}
+	/* We have data */
+	if (ret == 0) {
+		start_point_val = decode_key(cfg, key);
+		cursor->reset(cursor);
+		if ((ret = cursor->prev(cursor)) != 0) {
+			lprintf(cfg, ret, 0,
+			    "truncate setup end: prev failed");
+			return (-1);
```

Comment:
What does ```-1``` mean? You return it in a number of places. We already have a ```ret``` variable with more context - you should return that instead.

There are other places with similar style that discard a return from ```enomem```, then return ```-1```. Those should ```return (enomem(cfg));``` instead.